index.js

```diff
@@ -595,6 +607,40 @@ class XiaomiRoborockVacuum {
     }
   }
 
+  async getRoomList() {
+    await this.ensureDevice('getRoomList');
+
+    try {
+      const timers = await this.device.call('get_timer');
+
+      // Find specific timer containing the room order
+      // Timer needs to be scheduled for 00:00 and inactive
+      let leetTimer = timers.find(
+        x => x[2][0].startsWith("0 0") && x[1] == 'off');
+      if (leetTimer == undefined) {
+        this.log.error(`ERR getRoomList | ${this.model} | Could not find a timer for autoroom`);
+        return;
+      }
+
+      let roomIds = leetTimer[2][1][1]['segments'].split`,`.map(x=>+x);
+      this.log.debug(`DEB getRoomList | ${this.model} | Room IDs are ${roomIds}`);
+
+      if (roomIds.length != this.config.autoroom.length) {
+        this.log.error(`ERR getRoomList | ${this.model} | Number of rooms in config does not match number of rooms in the timer`);
+        return;
+      }
+      let roomMap = [];
+      for (const [i, roomId] of roomIds.entries()) {
+        this.rooms[i].id = roomId;
```

Comment:
rather than maintaining the rooms array, how about changing a property in the service?

```
const roomName = this.config.autoroom[i];
this.services[roomName].roomId = roomId;
```
Then you can refer to that instead of `this.setCleaningRoom(newState, this.rooms[idx].id), cb))`?
```
this.setCleaningRoom(newState, this.services[roomName].roomId), cb))
```

Reply:
Oh I didn’t know I could change the services afterwards, but you’re right, it works. Yes I guess this is a much nicer solution.